### ISSUE TITLE
Unable to bundle React into an AMD module

### ISSUE THREAD

**Issue by clarencejychan** [rollup/rollup#3254]
- Rollup Version: 1.17.0
- Operating System (or Browser): 78.0.3904.97
- Node Version: 10.15.0

### How Do We Reproduce?

i"m currently trying to bundle React into my code  so that there's a copy of react inside of my project (specific reason as to why we need this). The output is an AMD module that can be consumed by RequireJS. For the life of me I can't get requirejs recognizing my module as soon as I try to resolve any external dependencies. 
My config looks something like
```
import postcss from 'rollup-plugin-postcss'
import resolve from 'rollup-plugin-node-resolve'
import commonJS from 'rollup-plugin-commonjs'
export default {
  input: 'src/index.js', // source entry point
  output: {
    file: "dist/amd/mjolnir-dist.js",
    format: 'amd',
  },
  plugins: [
    resolve(),
    commonJS(),
    postcss({
      extract: true,
      sourceMap: true,
      extensions: ['.css'],
      plugins: []
    })
  ]
};
```

The output bundled file looks something like 
```
define(function () { 'use strict';
        ...
	var react = createCommonjsModule(function (module) {

	if (process.env.NODE_ENV === 'production') {
	  module.exports = react_production_min;
	} else {
	  module.exports = react_development;
	}
	});

       ....
	if (window.require) {
	  window.require.config({
	    map: {
	      '*': {
	        'mjolnir' : 'nbextensions/mjolnir/mjolnir'
	      }
	    }
	  });
	}

	function mjolnir() {
	  console.log("test");
	}

	return mjolnir;

});
```

https://github.com/rollup/rollup-plugin-commonjs/issues/114 seemed to have as similar issue.
### Expected Behavior
The module is consumed and runnable on the browser

### Actual Behavior
The script errors out during the define process and errors out with a 404. (See images and code snippet
![image (4)](https://user-images.githubusercontent.com/14989024/69459883-a0bf5200-0d27-11ea-8045-4d3e1efe290b.png)
) 

The AMD module that consumes this one does so in the following format
```
require.config({
  paths: {
    "dist": hostname + "/assets/mjolnir-dist" <------ The file in question
  }
});

define ([
  "dist"
], function (
    dist
) {
  'use strict';
  dist();
});
```

**Comment by shellscape** [comment]
@clarencejychan thanks for the issue, but you didn't follow instructions. Please add a proper reproduction and an actual problem description, or we'll close this as a support request.

**Comment by clarencejychan** [comment]
hi @shellscape, my use case is quite specific (also specific to my company) so it's a bit hard to follow repro steps. What would be the best way to give you guys more information? Should I just provide steps like:
```
`yarn add react`
etc.
then try to import it as an AMD module
```

Something along those lines?

**Comment by shellscape** [comment]
Please see the issue template you chose. It contains instructions.

We don't offer support here (per the issue template as well) so unfortunately if you  ant provide a reproduction and can't narrow down your error, we'll have to close as a support request.
